vtStor/Buffer.cpp

```diff
@@ -29,7 +29,12 @@ cBuffer::cBuffer( size_t SizeInBytes ) :
 
     // Allocate enough memory to ensure cache alignment
     m_UnalignedBuffer = new U8[ m_SizeInBytes + CACHE_ALIGN_BYTES ];
+#if _WIN32
     U16 currentAddress = ((U32)m_UnalignedBuffer & 0x0000FFFF);
+# endif
+#if __linux__
+    U16 currentAddress = static_cast<U32>(reinterpret_cast<intptr_t>(m_UnalignedBuffer)) & 0x0000FFFF; // add by Minh Mai static_cast<int>(reinterpret_cast<intptr_t>(p))
```

Comment:
On Linux, we cannot use regular cast look like on Win? And if the static_cast is the best solution, why we don't use it on Win. 
In my opinion, I don't want to see #if #endif here.